src/widgets/vitals/vitals-header/vital-header-state.component.tsx

```diff
@@ -0,0 +1,136 @@
+import { useConfig, useCurrentPatient } from "@openmrs/esm-react-utils";
+import React, { useEffect, useState } from "react";
+import {
+  PatientVitals,
+  performPatientsVitalsSearch
+} from "../vitals-card.resource";
+import styles from "./vital-header-state.component.scss";
+import VitalHeaderStateDetails from "./vital-header-details.component";
+import isToday from "dayjs/plugin/isToday";
+import VitalsHeaderStateTitle from "./vital-header-title.component";
+import dayjs from "dayjs";
+import { InlineLoading } from "carbon-components-react";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { isEmpty } from "lodash-es";
+dayjs.extend(isToday);
+interface viewState {
+  view: "Default" | "Warning";
+}
+
+const VitalHeader: React.FC = () => {
+  const config = useConfig();
+  const [, , patientUuid] = useCurrentPatient();
+  const [vitals, setVitals] = useState<PatientVitals>();
+  const [displayState, setDisplayState] = useState<viewState>({
+    view: "Default"
+  });
+  const [showDetails, setShowDetails] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const toggleView = () => setShowDetails(prevState => !prevState);
+
+  useEffect(() => {
+    if (patientUuid && config) {
+      const sub = performPatientsVitalsSearch(
+        config.concepts,
+        patientUuid
+      ).subscribe(
+        vitals => {
+          if (vitals.length) {
+            setVitals(vitals[0]);
```

Comment:
Coupled state. Should we have a more complex state object or even a reducer here? Once we need to use two (or more) `setX` instructions at the same time we know that something is not fully right.